referenceframe/frame_system.go

```diff
@@ -446,6 +387,57 @@ func StartPositions(fs FrameSystem) map[string][]Input {
 	return positions
 }
 
+// FrameSystemToPCD takes in a framesystem and returns a map where all elements are
+// the point representation of their geometry type with respect to the world.
+func FrameSystemToPCD(system FrameSystem, inputs map[string][]Input, logger golog.Logger) (map[string][]r3.Vector, error) {
+	vectorMap := make(map[string][]r3.Vector)
+	geometriesInWorldFrame, err := FrameSystemGeometries(system, inputs)
+	if err != nil {
+		logger.Debug(err)
+	}
+	for _, geometries := range geometriesInWorldFrame {
+		for _, geometry := range geometries.Geometries() {
+			vectorMap[geometry.Label()] = geometry.ToPoints(defaultPointDensity)
+		}
+	}
+	return vectorMap, nil
+}
+
+// FrameSystemGeometries takes in a framesystem and returns a map where all elements are GeometriesInFrames with a World reference frame.
+func FrameSystemGeometries(fs FrameSystem, inputMap map[string][]Input) (map[string]*GeometriesInFrame, error) {
+	var errAll error
+	allGeometries := make(map[string]*GeometriesInFrame, 0)
+	for _, name := range fs.FrameNames() {
+		geometries, err := FrameGeometriesInWorldFrame(fs, fs.Frame(name), inputMap)
+		if err != nil {
+			errAll = multierr.Append(errAll, err)
+			continue
+		}
+		allGeometries[name] = geometries
+	}
+	return allGeometries, errAll
+}
+
+// FrameGeometriesInWorldFrame returns the geometries associated with the given frame with respect to the frame system's World frame.
```

Comment:
This might be more useful if it transformed the given frame's geometries into any given frame, not just World